src/middlewares/error.ts

```diff
@@ -11,30 +12,34 @@ const zodErrorOptions: ErrorMessageOptions = {
   transform: ({ errorMessage, index }) => `Error #${index + 1}: ${errorMessage}`
 }
 
-const errorConverter = (err: any, sentry: Toucan) => {
+export const errorConverter = (err: unknown, sentry: Toucan): ApiError => {
   let error = err
   if (error instanceof ZodError) {
     const errorMessage = generateErrorMessage(error.issues, zodErrorOptions)
     error = new ApiError(httpStatus.BAD_REQUEST, errorMessage)
   } else if (!(error instanceof ApiError)) {
-    const statusCode = error.statusCode || httpStatus.INTERNAL_SERVER_ERROR
-    const message = error.description || error.message || httpStatus[statusCode]
+    const castedErr = (typeof error === 'object' ? error : {}) as Record<string, unknown>
+    const statusCode: number =
+      typeof castedErr.statusCode === 'number'
+        ? castedErr.statusCode
+        : httpStatus.INTERNAL_SERVER_ERROR
+    const message = (castedErr.description || castedErr.message || httpStatus[statusCode]) as string
     if (statusCode >= 500) {
       // Log any unhandled application error
       sentry.captureException(error)
     }
-    error = new ApiError(statusCode, message, false, err.stack)
+    error = new ApiError(statusCode, message, false)
   }
-  return error
+  return error as ApiError
 }
 
-const errorHandler: ErrorHandler<{ Bindings: Bindings }> = (err, c) => {
+export const errorHandler: ErrorHandler<{ Bindings: Bindings }> = (err, c) => {
   const config = getConfig(c.env)
   const sentry = getSentry(c)
   const error = errorConverter(err, sentry)
   if (config.env === 'production' && !error.isOperational) {
     error.statusCode = httpStatus.INTERNAL_SERVER_ERROR
-    error.message = httpStatus[httpStatus.INTERNAL_SERVER_ERROR]
+    error.message = httpStatus[httpStatus.INTERNAL_SERVER_ERROR].toString()
```

Comment:
The `httpStatus` object is defined kinda weirdly, meaning that some of the values are numbers, e.g.

```typescript
declare namespace httpStatus {
  interface HttpStatus {
    readonly [key: string]:
      | string
      | number
      | HttpStatusClasses
      | HttpStatusExtra;

    readonly 100: string;
    readonly '100_NAME': string;
    readonly '100_MESSAGE': string;
    readonly '100_CLASS': string;
    readonly CONTINUE: number;
    ...
```

So `httpStatus['CONTINUE']` would be a number, and even though typescript knows that `httpStatus.INTERNAL_SERVER_ERROR` is a number and all the number keys of `httpStatus` have string values, typescript can't figure out that the only values can be strings. So typescript thinks that it can either be a `string` or a `number`, therefore if we just call `.toString()` it solves the type issue.